File: src/lib/types/analytics.ts
export interface StatusBreakdownItem {
  status: string;
  count: number;
}

export interface MonthlyCountItem {
  month: string; // "YYYY-MM"
  count: number;
}

export interface AnalyticsSummary {
  totalAppointments: number;
  totalPatients: number;
  cancellationRate: number;
  statusBreakdown: StatusBreakdownItem[];
  patientsPerMonth: MonthlyCountItem[];
}

File: backend/app/schemas/analytics.py
from app.schemas.base import CamelModel


class StatusBreakdownItem(CamelModel):
    status: str
    count: int


class MonthlyCountItem(CamelModel):
    month: str  # "YYYY-MM"
    count: int


class AnalyticsSummaryOut(CamelModel):
    total_appointments: int
    total_patients: int
    cancellation_rate: float
    status_breakdown: list[StatusBreakdownItem]
    patients_per_month: list[MonthlyCountItem]
